### PR TITLE
[karma-discord-roles] feature: add karma-discord-roles

### PR DESCRIPTION
**Pull request by Arthh**
# karma-discord-roles

**- Karma Discord role strategy can be used by DAOs to provide voting power to contributors with a specific role on Discord.**

### CODE REVIEW

File: src/strategies/karma-discord-roles/index.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+  const votingPower = {};
+
+  const userAddresses = delegates.map((user) => user.publicAddress);
+  const paramAddresses = addresses.length ? addresses : [];
+  const allAddresses = [...userAddresses, ...paramAddresses];
+
+  allAddresses.forEach((address) => {
+    const checksumAddress = getAddress(address);
+    votingPower[checksumAddress] = 1;
+  });
```
Comment (ChaituVR) [inline src/strategies/karma-discord-roles/index.ts:43]: Hi @Arthh Did you try using `api-v2` strategy?

File: src/strategies/karma-discord-roles/index.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
+import fetch from 'cross-fetch';
+import { getAddress } from '@ethersproject/address';
+
+export const author = 'KarmaHq';
```
Comment (ChaituVR) [inline src/strategies/karma-discord-roles/index.ts:4]:
```suggestion
export const author = 'show-karma';
```

File: src/strategies/karma-discord-roles/index.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+export const version = '2.0.0';
```
Comment (ChaituVR) [inline src/strategies/karma-discord-roles/index.ts:5]:
```suggestion
export const version = '1.0.0';
```

**Review by ChaituVR — APPROVED** [review]

**Comment by zzuziak** [comment]
Hi team 👋  I might be wrong, but I can see @Arthh, you never use snapshot for the strategy. Is it giving someone voting power even if they have just acquired the role on Discord? One of our requirements is to calculate the VP based on the snapshot of proposal. cc @ChaituVR

**Comment by ChaituVR** [comment]
Correct, Informed them here https://discord.com/channels/707079246388133940/1126157732534288507/1130948591993761922 but discussing other changes.

**Pull request merged**